src/components/learner-credit-management/data/hooks/useCancelContentAssignments.js

```diff
@@ -0,0 +1,55 @@
+import { useCallback, useState } from 'react';
+import { logError } from '@edx/frontend-platform/logging';
+import { useToggle } from '@edx/paragon';
+import EnterpriseAccessApiService from '../../../../data/services/EnterpriseAccessApiService';
+
+const generateSuccessCancelMessage = (assignmentUuids) => {
+  if (Array.isArray(assignmentUuids)) {
+    return `Assignments canceled (${assignmentUuids.length})`;
+  }
+
+  return 'Assignment canceled';
+};
+
+const useCancelContentAssignments = (
+  assignmentConfigurationUuid,
+  refresh,
+  tableInstance,
+  uuids,
+) => {
+  const [isOpen, open, close] = useToggle(false);
+  const [showToast, setShowToast] = useState(false);
+  const [toastMessage, setToastMessage] = useState('');
+
+  const cancelContentAssignments = useCallback(async () => {
+    const handleRefresh = () => {
+      refresh(tableInstance);
+    };
+    try {
+      const { status } = await EnterpriseAccessApiService.cancelContentAssignments(assignmentConfigurationUuid, uuids);
+      if (status === 200) {
+        setToastMessage(generateSuccessCancelMessage(uuids));
+        setShowToast(true);
+        close(true);
+        setTimeout(() => {
+          handleRefresh();
+        }, 2000);
+      }
+    } catch (err) {
+      logError(err);
+      close(true);
```

Comment:
nit: if both the `try` and `catch` blocks both call `close`, I'm wondering if it'd make sense to put that in `finally` block to save calling it both places. 

nit/sanity check: what does the `close` function do with the boolean argument? is it needed?

Reply:
Removed the closed function from try catch blocks